dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/DependencyConflict.java

```diff
@@ -62,18 +62,35 @@ public DependencyPath getPathToSelectedArtifact() {
   @Override
   public String toString() {
     Artifact selected = pathToSelectedArtifact.getLeaf();
+    String selectedCoordinates = Artifacts.toCoordinates(selected);
     Artifact unselected = pathToArtifactThruSource.getLeaf();
-    return "Dependency conflict: "
-        + Artifacts.toCoordinates(selected)
-        + " does not define "
-        + symbol
-        + " but "
-        + Artifacts.toCoordinates(unselected)
-        + " defines it.\n"
-        + "  selected: "
-        + pathToSelectedArtifact
-        + "\n  unselected: "
-        + pathToArtifactThruSource;
+    String unselectedCoordinates = Artifacts.toCoordinates(unselected);
+
+    if (linkageProblem instanceof AbstractMethodProblem) {
```

Comment:
Anytime I see instanceof I assume it can be replaced with polymorphism. This can be a single method in LinkageProblem.

Reply:
I'm feeling that's unnecessary abstraction to remove just one if-statement. Let me try what it looks like.